terraform/example/server.tf.example

```diff
@@ -0,0 +1,11 @@
+resource "teleport_server" "ssh_agentless" {
+    version = "v2"
+    sub_kind = "openssh"
+    metadata = {
+        name = "test"
```

Comment:
Did you test this?

I was testing for the `openssh-ec2-ice` nodes and it doesn't work if we give it a non-uuid string.

An example that works:
```tf
resource "random_uuid" "ec2-random-name" {
}

resource "teleport_server" "my-dev-server" {
  version  = "v2"
  sub_kind = "openssh-ec2-ice"
  metadata = {
    name = "${random_uuid.ec2-random-name.result}"
  }
  spec = {
    addr     = "<ec2 private ip>:22"
    hostname = "my-dev-server"
    cloud_metadata = {
      aws = {
        account_id  = "<account-id>"
        instance_id = "<instance-id>"
        region      = "<region>"
        vpc_id      = "<vpc-id>"
        integration = "<awsoidc-integration-used-for-credentials>"
        subnet_id   = "<subnet-id>"
      }
    }
  }
}

```

Reply:
> Did you test this?
> 
> I was testing for the openssh-ec2-ice nodes and it doesn't work if we give it a non-uuid string.

Do you mean against a real AWS account? No, I only checked that Teleport was accepting the resource in the operator. I don't understand why the eice server name must be a uuid 🤔

Reply:
It's a bit buried, but it is required to be an uuid because of this
https://github.com/gravitational/teleport/blob/a39c21716b40518f0433b619bb9828d127bd6b77/api/utils/route.go#L41

When I implemented EC2 ICE mode, I closely followed the OpenSSH/Agentless mode.
And I ended up making the same assumptions.

Reply:
I guess you shouldn't set the name. Instead, you should set the hostid

Reply:
I updated the TF example to not set the name, and added tests for both named and nameless openssh servers.